lib/Sema/Constraint.cpp

```diff
@@ -722,6 +722,23 @@ Constraint *Constraint::createDisjunction(ConstraintSystem &cs,
     return constraints.front();
   }
 
+#ifndef NDEBUG
+  assert(!constraints.empty());
+  // Verify that all disjunction choices have the same left-hand side.
+  Type commonType;
+  assert(llvm::all_of(constraints, [&](const Constraint *choice) -> bool {
+    auto currentType = choice->getFirstType()->lookThroughAllOptionalTypes();
+    assert(currentType);
```

Comment:
This assert doesn't seem all that useful since we would crash in `lookThroughAllOptionalTypes()` if `getFirstType()` returned false and `lookThroughAllOptionalTypes()` never returns `nullptr` (and it would break code all over if that changed).

Reply:
Ah, right, just went overly paranoid... :)

Reply:
Removed assert and made it so we skip disjunctions formed by restrictions/fixes, this way we don't have to `lookThroughAllOptionalTypes`